crates/provider/src/kubernetes/client/kube_rs.rs

```diff
@@ -0,0 +1,379 @@
+use std::{
+    collections::BTreeMap, fmt::Debug, os::unix::process::ExitStatusExt, path::Path,
+    process::ExitStatus,
+};
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use futures::{StreamExt, TryStreamExt};
+use k8s_openapi::api::core::v1::{ConfigMap, Namespace, Pod, PodSpec};
+use kube::{
+    api::{AttachParams, DeleteParams, ListParams, LogParams, PostParams, WatchParams},
+    core::{ObjectMeta, WatchEvent},
+    Api, Client, Resource,
+};
+use serde::de::DeserializeOwned;
+use sha2::digest::Digest;
+use support::fs::FileSystem;
+use tokio::io::{AsyncRead, AsyncReadExt, AsyncWriteExt};
+use tokio_util::compat::FuturesAsyncReadCompatExt;
+
+use crate::types::ExecutionResult;
+
+use super::KubernetesClient;
+
+pub struct KubeRsKubernetesClient<FS>
+where
+    FS: FileSystem + Send + Sync + Clone,
+{
+    client: kube::Client,
+    filesystem: FS,
+}
+
+impl<FS> KubeRsKubernetesClient<FS>
+where
+    FS: FileSystem + Send + Sync + Clone,
+{
+    async fn new(filesystem: FS) -> Result<Self, Box<dyn std::error::Error>> {
+        Ok(Self {
+            // TODO: make it more flexible with path to kube config
```

Comment:
I think is ok for now, I saw most of the time usage of the default location of the config.

Reply:
Yes, was just an idea for a specific use case, can be let here and in the future if we have time to make things cleaner, it's more a "idea list"